src/pages/index.js

```diff
@@ -38,30 +94,80 @@ cardAddButton.addEventListener("click", () => {
   cardAddPopup.open();
 });
 
+avatarEditBtn.addEventListener("click", () => {
+  avatarEditPopup.open();
+});
+
 function renderCard(cardData) {
   const newCard = getCardElement(cardData);
   cardList.addItem(newCard);
 }
 
-const cardList = new Section(
-  { items: initialCards, renderer: renderCard },
-  ".cards__list"
-);
+const cardList = new Section({ renderer: renderCard }, ".cards__list");
 
-cardList.renderItems();
+api
+  .getAppInfo()
+  .then(([userData, cards]) => {
+    cardList.renderItems(cards);
+    profileUserInfo.setUserInfo(userData);
+    profileUserInfo.setUserAvatar(userData);
+  })
+  .catch((error) => console.error(error));
 
 const cardAddPopup = new PopupWithForm({
   popupSelector: "#card-add-modal",
   handleFormSubmit: (data) => {
-    renderCard(data);
-    cardFormValidator.disableButton();
+    renderLoading("#card-add-modal", true);
+    api
+      .addCard(data)
+      .then((card) => {
+        renderCard(card);
+        cardFormValidator.disableButton();
+      })
+      .catch((error) => {
+        console.log(error);
+      })
+      .finally(() => {
+        renderLoading("#card-add-modal", false);
+      });
   },
 });
 
 const profileEditPopup = new PopupWithForm({
   popupSelector: "#profile-edit-modal",
-  handleFormSubmit: (data) => {
-    profileUserInfo.setUserInfo(data);
+  handleFormSubmit: ({ title, description }) => {
+    renderLoading("#profile-edit-modal", true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```